src/isar/models/mission.py

```diff
@@ -1,37 +1,60 @@
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 from datetime import datetime
-from typing import List, Union
-from uuid import UUID
+from typing import Iterator, List, Union
+from uuid import UUID, uuid4
 
 from isar.config.settings import settings
 from isar.models.mission_metadata.mission_metadata import MissionMetadata
-from robot_interface.models.mission import STEPS
-from robot_interface.models.mission.step import DriveToPose, InspectionStep
+from robot_interface.models.mission import STEPS, Step, StepStatus
 
 
 @dataclass
-class Mission:
+class Task:
     steps: List[STEPS]
+    status: StepStatus = field(default=StepStatus.NotStarted, init=False)
+    id: UUID = field(default_factory=uuid4, init=False)
+    iterator: Iterator = None
+
+    def next_step(self) -> Step:
+        return self.iterator.__next__()
+
+    def is_finished(self) -> bool:
+        for step in self.steps:
+            if step.status is StepStatus.Failed or step.status is StepStatus.Unexpected:
```

Comment:
Remove `or step.status is StepStatus.Unexpected:` ? This case is covered by the elif below.

Reply:
Thank you!